test/unit-test/mqtt_agent_command_functions_utest.c

```diff
@@ -73,3 +170,265 @@ int suiteTearDown( int numFailures )
 {
     return numFailures;
 }
+
+/* ========================================================================== */
+
+/**
+ * @brief Test that MQTTAgentCommand_ProcessLoop() works as intended.
+ */
+void test_MQTTAgentCommand_ProcessLoop( void )
+{
+    MQTTAgentContext_t mqttAgentContext;
+    void * pUnusedArg;
+    MQTTAgentCommandFuncReturns_t returnFlags;
+    MQTTStatus_t mqttStatus;
+
+    mqttStatus = MQTTAgentCommand_ProcessLoop( &mqttAgentContext, pUnusedArg, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+}
+
+/**
+ * @brief Test MQTTAgentCommand_Publish() success case with QoS0.
+ */
+void test_MQTTAgentCommand_Publish_QoS0_success( void )
+{
+    MQTTAgentContext_t mqttAgentContext;
+    MQTTPublishInfo_t publishInfo;
+    MQTTAgentCommandFuncReturns_t returnFlags;
+    MQTTStatus_t mqttStatus;
+
+    /* Initializing QOS. */
+    publishInfo.qos = MQTTQoS0;
+
+    MQTT_Publish_IgnoreAndReturn( MQTTSuccess );
+
+    mqttStatus = MQTTAgentCommand_Publish( &mqttAgentContext, &publishInfo, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+}
+
+/**
+ * @brief Test MQTTAgentCommand_Publish() success case with QoS1.
+ */
+void test_MQTTAgentCommand_Publish_QoS1_success( void )
+{
+    MQTTAgentContext_t mqttAgentContext;
+    MQTTPublishInfo_t publishInfo;
+    MQTTAgentCommandFuncReturns_t returnFlags;
+    MQTTStatus_t mqttStatus;
+
+    /* Initializing QOS. */
+    publishInfo.qos = MQTTQoS1;
+
+    MQTT_GetPacketId_IgnoreAndReturn( 1 );
+    MQTT_Publish_IgnoreAndReturn( MQTTSuccess );
```

Comment:
Instead of Ignore and Return, we can start using Expect and Return. We know all the arguments the publish should be called with so let's use that here